programs/bpf_loader/src/syscalls/mod.rs

```diff
@@ -3497,7 +3558,33 @@ mod tests {
             clean_epochschedule.warmup = src_epochschedule.warmup;
             clean_epochschedule.first_normal_epoch = src_epochschedule.first_normal_epoch;
             clean_epochschedule.first_normal_slot = src_epochschedule.first_normal_slot;
-            assert!(are_bytes_equal(&got_epochschedule, &clean_epochschedule));
+            assert!(are_bytes_equal(
+                &got_epochschedule_obj,
+                &clean_epochschedule
+            ));
+
+            let result = SyscallGetSysvar::rust(
+                &mut invoke_context,
+                epochschedule_id_va,
+                got_epochschedule_buf_va,
+                0,
+                EpochSchedule::size_of() as u64,
+                0,
+                &mut memory_mapping,
+            );
+            result.unwrap();
+
+            // clone is to zero the alignment padding
+            let epochschedule_from_buf =
+                bincode::deserialize::<EpochSchedule>(&got_epochschedule_buf)
+                    .unwrap()
+                    .clone();
+
+            assert_eq!(epochschedule_from_buf, src_epochschedule);
+            assert!(are_bytes_equal(
+                &epochschedule_from_buf,
+                &clean_epochschedule
+            ));
```

Comment:
Just checking: This first assertion would pass without the clone, right? It's only to have the bytes equal? Just trying to think how we need to handle padding in the API that we expose to programs.

Reply:
yes thats right, the objects are equal by the rules of rust, and the clone is just for comparing the memory mapping using `unsafe`. ill move the clone down so thats explicit

everything presently exposed to programs goes through `get_sysvar`, so theres no change:
```
    let var = translate_type_mut::<T>(memory_mapping, var_addr, check_aligned)?;

    // this clone looks unecessary now, but it exists to zero out trailing alignment bytes
    // it is unclear whether this should ever matter
    // but there are tests using MemoryMapping that expect to see this
    // we preserve the previous behavior out of an abundance of caution
    let sysvar: Arc<T> = sysvar?;
    *var = T::clone(sysvar.as_ref());

    Ok(SUCCESS)
```

when we transition to the new syscall, i would just alloc a buffer that matches `mem::size_of` and `serialize_into` it, this only happens because `bincode::deserialize` doesnt zero its destination